Improve `RedundantInheritedCheck` (#272)
integrated-application-development/sonar-delphi: CHANGELOG.md

- Improve AST modeling around property specifiers.
- Improve semantic analysis around property specifiers.
- Exclude properties with `implements` specifiers in `UnusedProperty`.

> - Improve `RedundantInheritedCheck`.
> 
> ### Fixed
> 
> - Overly permissive parsing rules around `string` and `file` types.
> - `RedundantInheritedCheck` removing comments and compiler directives with quick fixes.

Reviewer: ```suggestion

### Fixed

- Overly permissive parsing rules around `string` and `file` types.
- False positives around `message` handler methods in `RedundantInheritedCheck`.
- Quick fixes removing comments and compiler directives in `RedundantInheritedCheck`.
```

### Fixed

- Overly permissive parsing rules around `string` and `file` types.

> - Quick fixes removing comments and compiler directives in `RedundantInheritedCheck`.
> - False positives around `message` handler methods in `RedundantInheritedCheck`.

Reviewer: Oops, missed one thing. The rule key doesn't include the word `Check`.

```suggestion
- Quick fixes removing comments and compiler directives in `RedundantInherited`.
- False positives around `message` handler methods in `RedundantInherited`.
```

- Overly permissive parsing rules around `string` and `file` types.
- Quick fixes removing comments and compiler directives in `RedundantInherited`.

> - False positives around `message` handler methods in `RedundantInherited`.

Reviewer: ```suggestion
- False positives around `message` handler methods in `RedundantInherited`.
- False positives when parameter names were mismatched to the inherited method in `RedundantInherited`.
```
